spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/elasticsearch/rest/RestClientAutoConfigurationTests.java

```diff
@@ -74,6 +76,31 @@ public void configureWhenBuilderCustomizerShouldApply() {
 				});
 	}
 
+	@Test
+	public void defaultTimeoutsShouldBeConfigured() {
+		this.contextRunner.run((context) -> {
+			assertThat(context).hasSingleBean(RestClient.class);
+			assertThat(context).hasSingleBean(RestClientProperties.class);
+			RestClient restClient = context.getBean(RestClient.class);
+			RestClientProperties properties = context.getBean(RestClientProperties.class);
+			assertTimeouts(restClient, properties.getReadTimeout(),
```

Comment:
That's not what that test is supposed to do. It should test that when no defaults are set, the defaults in the properties object are consistent. 

So you should configure a `RestClient` manually (i.e. without the auto-config) and query its timeouts and validate they are the same as the one defined in the configuration by default to ensure our defaults match the ones of ES if that's the plan.